feat: apis for attaching chatlogs and polls to session materials

rjsparks opened the pull request:
Very early look into where this is going, mostly so @alexamirante can comment on what the apis want to ingest.

Still much to be done - tests, initial views of the documents once we have them, etc.

rjsparks committed:
feat: apis for attaching chatlogs and polls to session materials

jennifer-richards commented on ietf/meeting/utils.py:
```diff
@@ -710,3 +710,34 @@ def handle_upload_file(file, filename, meeting, subdir, request=None, encoding=N
+def new_doc_for_session(type_id, session):
+    typename = DocTypeName.objects.get(slug=type_id)
+    ota = session.official_timeslotassignment()
+    sess_time = ota and ota.timeslot.time
```
We'll need to make this TZ aware - or we could do it now like this:

```suggestion
    sess_time = ota and ota.timeslot.local_start_time()
```

Also, you'll need to handle the case where `ota is None` a few more times since it'll lead to `sess_time` being `None` in the code that follows.

rjsparks replied:
hmm - it would be very odd for ota to be None since this is used to report on a session that _happened_ already (and was set up well enough to happen in MeetEcho), but I'll look into better resiliency.

jennifer-richards replied:
It just struck me as odd that there's the guard in this line.

jennifer-richards reviewed (commented):
Leafed through this. One detailed comment inline. API looks ok to me but it'd be nice to document the expected structure.

We're starting to see a lot of these boilerplate-ful API endpoints. Would be really nice to find a library that we could lean on instead of reinventing this each time. It'd be nice if it let us return text/plain or application/json based on the request headers, too, so that we don't hard code in that we speak JSON one way and text/plain the other.

rjsparks committed:
fix: anticipate becoming tzaware, and improve guard against attempts to provide docs for sessions that have no official timeslot assignment.

rjsparks committed:
fix: get chatlog upload to actually work

Modifications to several initial implementation decisions.
Updates to the fixtures.

rjsparks committed:
fix: test polls upload

Refactored test to reduce duplicate code

rjsparks committed:
fix: allow api keys to be created for the new endpoints

rjsparks committed:
feat: add ability to view chatlog and polls documents. Show links in session materials.

rjsparks committed:
fix: commit new template

jennifer-richards commented on ietf/doc/views_doc.py:
```diff
@@ -725,6 +724,26 @@ def document_main(request, name, rev=None):
                            assignments=assignments,
                       ))
 
+    if doc.type_id == "chatlog":
```
Is a "polls" clause coming, too?

rjsparks replied:
yes, thanks for the nudge.

jennifer-richards reviewed (commented):
Had another look. Things look reasonable, a couple little things I noticed/wondered.

rjsparks commented on ietf/name/fixtures/names.json:
```diff
@@ -3345,7 +3444,7 @@
       "is_schedulable": false,
       "material_types": "[\n    \"slides\"\n]",
-      "matman_roles": "[]",
+      "matman_roles": "[\n    \"chair\"\n]",
```
yes, and it's ok (or at least mostly unavoidable) - that reflects a change that was made in production using the admin.

rjsparks commented on ietf/doc/migrations/0045_docstates_chatlogs_polls.py:
```diff
@@ -0,0 +1,34 @@
+# Copyright The IETF Trust 2022, All Rights Reserved
+from django.db import migrations
+
+def forward(apps, shema_editor):
```
will fix

rjsparks replied:
done

rjsparks committed:
fix: typo in migration signatures

rjsparks committed:
feat: add main doc page handling for polls. Improve tests.

rjsparks committed:
Merge branch 'main' into chat-and-polls

rjsparks commented:
This is now only waiting for some better FE rendering of the chatlog and polls documents

NGPixel committed:
feat: chat log vue component + embedded vue loader

rjsparks committed:
Merge branch 'ietf-tools:main' into chat-and-polls

rjsparks committed:
feat: render polls using Vue

rjsparks committed:
fix: address pug syntax review comments from Nick.

rjsparks committed:
fix: repair remaining mention of chat log from copymunging

rjsparks committed:
fix: use double-quotes in html attributes

rjsparks committed:
fix: provide missing choices update migration

NGPixel committed:
test: silence html validator empty attr warnings

NGPixel committed:
test: fix test_runner config

rjsparks committed:
fix: locate session when looking at a dochistory object for polls or chatlog
